tests/BuildScriptGenerator.Tests/DotnetCore/GlobalJsonSdkResolverTest.cs

```diff
@@ -0,0 +1,1217 @@
+            var availableVersions = new[] { "3.1.100", "3.1.101", "3.1.102-preview1-03444", "3.1.102-preview2-03444" };
+            var globalJson = new GlobalJsonModel();
+            globalJson.Sdk = new Sdk
+            {
+                Version = "3.1.102-preview1-03444",
+                RollForward = GlobalJsonSdkResolver.Disable,
+                AllowPreRelease = allowPrerelease,
+            };
+            var globalJsonHelper = GetGlobalJsonHelper();
+
+            // Act
+            var actual = globalJsonHelper.GetSatisfyingSdkVersion(globalJson, availableVersions);
+
+            // Assert
+            Assert.Equal(expectedVersion, actual);
+        }
+
+        [Fact]
+        public void Disable_ReturnsNull_IfExactVersionIsNotFound()
+        {
+            // Arrange
+            var availableVersions = new[] { "2.1.100", "3.1.101" };
+            var globalJson = new GlobalJsonModel();
+            globalJson.Sdk = new Sdk
+            {
+                Version = "3.1.100",
+                RollForward = GlobalJsonSdkResolver.Disable,
+            };
+            var globalJsonHelper = GetGlobalJsonHelper();
+
+            // Act
+            var actual = globalJsonHelper.GetSatisfyingSdkVersion(globalJson, availableVersions);
+
+            // Assert
+            Assert.Null(actual);
+        }
+
+        [Fact]
+        public void Patch_ReturnsSameVersion_IfPresent_AndDoesNotGetLatestPatchVersion()
+        {
+            // Arrange
+            var expectedVersion = "3.1.100";
+            var availableVersions = new[] { "2.1.100", "3.1.100", "3.1.101" };
+            var globalJson = new GlobalJsonModel();
+            globalJson.Sdk = new Sdk
+            {
+                Version = "3.1.100",
+                RollForward = GlobalJsonSdkResolver.Patch,
+            };
+            var globalJsonHelper = GetGlobalJsonHelper();
+
+            // Act
+            var actual = globalJsonHelper.GetSatisfyingSdkVersion(globalJson, availableVersions);
+
+            // Assert
+            Assert.Equal(expectedVersion, actual);
+        }
+
+        [Fact]
+        public void Patch_ReturnsLatestPatchVersion_IfSpecifiedPatchDoesNotExist()
+        {
+            // Arrange
+            var expectedVersion = "3.1.102";
+            var availableVersions = new[] { "2.1.100", "3.1.101", "3.1.102", "3.2.103" };
+            var globalJson = new GlobalJsonModel();
+            globalJson.Sdk = new Sdk
+            {
+                Version = "3.1.100",
+                RollForward = GlobalJsonSdkResolver.Patch,
+            };
+            var globalJsonHelper = GetGlobalJsonHelper();
+
+            // Act
+            var actual = globalJsonHelper.GetSatisfyingSdkVersion(globalJson, availableVersions);
+
+            // Assert
+            Assert.Equal(expectedVersion, actual);
+        }
+
+        [Fact]
+        public void Patch_ReturnsNull_IfSpecifiedPatchOrLatestPatchVersion_DoesNotExist()
```

Comment:
nit: `Patch_ReturnsNull_IfSpecifiedPatchOrLatestPatchVersionDoesNotExist`

Reply:
I think currently this test name follows the format you suggested `TestFeature_ReturnsWhat_IfWhatHappens_AndDoesNotDoOtherThings`